ext/mbstring/mbstring.c

```diff
@@ -1717,33 +1717,22 @@ PHP_FUNCTION(mb_str_split)
 
 static size_t mb_get_strlen(zend_string *string, const mbfl_encoding *encoding)
 {
-	size_t len = 0;
-
-	if (encoding->flag & MBFL_ENCTYPE_SBCS) {
-		return ZSTR_LEN(string);
-	} else if (encoding->flag & MBFL_ENCTYPE_WCS2) {
-		return ZSTR_LEN(string) / 2;
-	} else if (encoding->flag & MBFL_ENCTYPE_WCS4) {
-		return ZSTR_LEN(string) / 4;
-	} else if (encoding->mblen_table) {
-		const unsigned char *mbtab = encoding->mblen_table;
-		unsigned char *p = (unsigned char*)ZSTR_VAL(string), *e = p + ZSTR_LEN(string);
-		while (p < e) {
-			p += mbtab[*p];
-			len++;
-		}
+	unsigned int char_len = encoding->flag & (MBFL_ENCTYPE_SBCS | MBFL_ENCTYPE_WCS2 | MBFL_ENCTYPE_WCS4);
+	if (char_len) {
+		return ZSTR_LEN(string) / char_len;
 	} else {
 		uint32_t wchar_buf[128];
 		unsigned char *in = (unsigned char*)ZSTR_VAL(string);
 		size_t in_len = ZSTR_LEN(string);
 		unsigned int state = 0;
+		size_t len = 0;
 
 		while (in_len) {
 			len += encoding->to_wchar(&in, &in_len, wchar_buf, 128, &state);
 		}
-	}
 
-	return len;
+		return len;
+	}
```

Comment:
Could remove the ``else`` block entirely.

Also, the commit message for the first commit seems to be missing the benchmark times for the short strings.

Reply:
Let me remove that `else` block.

I'm a bit confused by the statement that benchmark times are missing; they're right there.

```
    EUC-KR, short (0-5 chars)        - master faster by 11.90% (0.0000 vs 0.0000)
    EUC-JP, short (0-5 chars)        - master faster by 10.88% (0.0000 vs 0.0000)
    BIG-5, short (0-5 chars)         - master faster by 10.66% (0.0000 vs 0.0000)
    UTF-8, short (0-5 chars)         - master faster by 8.91% (0.0000 vs 0.0000)
    CP936, short (0-5 chars)         - master faster by 6.27% (0.0000 vs 0.0000)
    UHC, short (0-5 chars)           - master faster by 5.38% (0.0000 vs 0.0000)
    SJIS, short (0-5 chars)          - master faster by 5.20% (0.0000 vs 0.0000)
```

Reply:
Yes, but they are all ``0.0000`` and then having a % saying how faster they are seems very misleading :)